components/table/Column.razor.cs

```diff
@@ -191,8 +191,11 @@ private void SetSorter(SortDirection sortDirection)
 
         private void ToggleTreeNode()
         {
+            bool expandValueBeforeChange = RowData.Expanded;
             RowData.Expanded = !RowData.Expanded;
-            Table?.Refresh();
+            Table?.OnExpand(RowData.CacheKey);
```

Comment:
Should it be invoked when it was expanded or not?

Reply:
The name suggests that it should be invoked on expand only. However I do not see the reason why only then. Since the consumer will get the "state" of expanded in `RowData.Expanded` (the event is passing `RowData`), then the consumer may decide which event variant he/she is interested in. And actually, in my case, I am interested in both - when the expanded is showing and when it is hiding.
Maybe we should change the event name to "OnExpandClicked" or alike?

Reply:
How about `OnExpandChange` ?

Reply:
Sure, no problem 👍🏻 . When I was adding it, I based it on antD, but you are right, this is better. There is always a dilemma - follow the antD or deviate. Regardless, I am glad you are ok with that solution.